### ISSUE TITLE
Equal duration checks inconsistent with float behaviour

### ISSUE THREAD

**Issue by terrorfisch** [qutech/qupulse#533]
```python
from qupulse.pulses import FunctionPT, MappingPT, AtomicMultiChannelPT

seconds2ns=1e9
pulse_duration = 1.0765001496284785e-07

fpt_1 = FunctionPT('sin(t)','duration', 'CHA')
fpt_2 = FunctionPT('cos(t)', pulse_duration * seconds2ns, 'CHB')
mpt = MappingPT.from_tuple((fpt_1, {'duration': seconds2ns * pulse_duration,}))


full_template = AtomicMultiChannelPT(mpt, fpt_2)
```

fails with `ValueError: Could not assert duration equality of 107.650014962848 and 107.650014962848`

**Comment by terrorfisch** [comment]
Direct cause of this is an error in `MappingPulseTempalte.duration`

**Comment by terrorfisch** [comment]
Fixed in #534 and #603
